### PR TITLE
Allow specifying a GCP project for E2E tests using a flag

### PR DESCRIPTION
**Pull request by adrcunha**
`--gcp-project` works locally or on Prow, `$PROJECT_ID` still works locally.

### CODE REVIEW

File: scripts/e2e-tests.sh [hunk 1]
```diff
@@ -319,22 +331,41 @@ function initialize() {
...
     esac
     shift
   done
+
+  # Use PROJECT_ID if set, unless --gcp-project was used.
+  if [[ -n "${PROJECT_ID:-}" && -z "${GCP_PROJECT}" ]]; then
+    echo "\$PROJECT_ID is set to '${PROJECT_ID}', using it to run the tests"
+    GCP_PROJECT="${PROJECT_ID}"
+  fi
+  if (( ! IS_PROW )) && [[ -z "${GCP_PROJECT}" ]]; then
```
Comment (srinivashegde86) [inline scripts/e2e-tests.sh]: should this be IS_BOSKOS as well?

File: scripts/e2e-tests.sh [hunk 2]
```diff
@@ -45,6 +45,9 @@ readonly E2E_CLUSTER_NODES=3
 readonly E2E_CLUSTER_MACHINE=n1-standard-4
 readonly TEST_RESULT_FILE=/tmp/${E2E_BASE_NAME}-e2e-result
 
+# Flag whether test is using a boskos GCP project
+IS_BOSKOS=0
```
Comment (srinivashegde86) [inline scripts/e2e-tests.sh]: when will this be not 0?

File: scripts/e2e-tests.sh (hunk 1 above)
Comment (adrcunha) [reply scripts/e2e-tests.sh]: No. IS_BOSKOS is set later.

File: scripts/e2e-tests.sh (hunk 2 above)
Comment (adrcunha) [reply scripts/e2e-tests.sh]: See line 362.

**Pull request merged**